src/Psy/Command/SandboxCommand.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace Psy\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class SandboxCommand
+ * @package Psy\Command
+ */
+class SandboxCommand extends Command
+{
+    const SANDBOX_FOLDER = 'psysh_sandbox';
+
+    const ACTION_ADD = 'add';
+    const ACTION_LIST = 'list';
+    const ACTION_WHICH = 'which';
+    const ACTION_SWITCH = 'switch';
+    const ACTION_DELETE = 'delete';
+    const ACTION_EXIT = 'exit';
+
+    /**
+     * @var array
+     */
+    protected $allowedActions = array(
+        self::ACTION_ADD,
+        self::ACTION_LIST,
+        self::ACTION_WHICH,
+        self::ACTION_SWITCH,
+        self::ACTION_DELETE,
+        self::ACTION_EXIT,
+    );
+
+    /**
+     * @var array
+     */
+    protected $sandboxes = array();
+
+    /**
+     * @var String
+     */
+    protected $restoreFolder;
+
+    /**
+     * @var string
+     */
+    protected $current;
+
+    /**
+     * @var InputInterface
+     */
+    protected $input;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('sandbox')
+            ->setDefinition(array(
+                new InputArgument(
+                    'action',
+                    InputArgument::REQUIRED,
+                    'Action to perform over the sandboxes add|list|switch|delete|exit'
+                ),
+                new InputArgument(
+                    'name',
+                    InputArgument::OPTIONAL,
+                    'Optional name for the sandbox'
+                ),
+            ))
+            ->setDescription('Manages sandboxed environments.')
+            ->setHelp(
+                <<<HELP
+sandbox add [name] creates a encapsulated sandbox with an optional name, if not provided will autogenerate a name.
+sandbox list Will show a list of the currently generated sandboxes
+sandbox which Will show the current sandbox
+sandbox switch name Switches to sandbox by name,
+sandbox delete [name] Removes a sandbox by name, or the current if name is not provided.
+sandbox exit Erases all sandboxes and sets the normal state.
+HELP
+            );
+
+        $this->restoreFolder = exec('pwd');
```

Comment:
possibly store this on the first call to use a sandbox, rather than when the command is created?